src/constitutive/mechanics/solid/small_strain_J2_plasticity_damage.cpp

```diff
@@ -63,186 +63,228 @@ void small_strain_J2_plasticity_damage::update_internal_variables(double const t
         auto& energy_var = energy_release_rates[l];
 
         // Elastic stress predictor
-        cauchy_stress = compute_cauchy_stress(material.shear_modulus(),
-                                              material.lambda(),
-                                              strain - plastic_strain);
+        cauchy_stress = compute_cauchy_stress_scalar_damage(material.shear_modulus(),
+                                                            material.lambda(),
+                                                            strain - plastic_strain,
+                                                            scalar_damage);
 
-        matrix3 tau = deviatoric(cauchy_stress) / (1.0 - scalar_damage) - deviatoric(back_stress);
+        // Trial state
+        matrix3 tau = cauchy_stress - back_stress;
```

Comment:
Is it possible to change that from `tau` to something like `trial_state` or `trial_stress`?

Reply:
I've just found what it should be called "relative stress"

Reply:
Done. I added one extra function to compute the energy release rate which makes it easier to introduce crack-closure effect later if needed.